src/com/google/enterprise/adaptor/database/UniqueKey.java

```diff
@@ -239,9 +243,8 @@ String makeUniqueId(ResultSet rs, boolean encode) throws SQLException {
           part = "" + rs.getLong(name);
           break;
         default:
-          String errmsg = "Invalid type '" + types.get(name) + "' for '"
-              + name + "'. Valid types are: int, string, timestamp, date"
-              + ", time and long.";
+          String errmsg = "Unknown column type for '" + name + "'. "
```

Comment:
A null argument to switch throws a NPE. I actually think omitting the default on an enum switch is reasonable (and it's permitted by the style guide).